src/prototype_creep_resources.js

```diff
@@ -486,7 +486,7 @@ Creep.prototype.transferMy = function() {
 };
 
 Creep.prototype.getEnergy = function() {
-  if (this.carry.energy == this.carryCapacity) {
+  if (this.carry.energy > 0) {
```

Comment:
I don't think this will work. `getEnergy` is actually used to `harvest` energy from sources. In that case the harvester would only get 1 tick energy from the source.